singularity-opac/neutrinos/non_cgs_neutrinos.hpp

```diff
@@ -0,0 +1,189 @@
+// ======================================================================
+// © 2021. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+// ======================================================================
+
+#ifndef SINGULARITY_OPAC_NEUTRINOS_NON_CGS_NEUTRINOS_
+#define SINGULARITY_OPAC_NEUTRINOS_NON_CGS_NEUTRINOS_
+
+#include <cassert>
+#include <cmath>
+#include <cstdio>
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-opac/base/opac_error.hpp>
+
+namespace singularity {
+namespace neutrinos {
+
+template <typename Opac>
+class NonCGSUnits {
+ public:
+  NonCGSUnits() = default;
+  NonCGSUnits(Opac &&opac, const Real time_unit, const Real mass_unit,
+              const Real length_unit, const Real temp_unit)
+      : opac_(std::forward<Opac>(opac)), time_unit_(time_unit),
+        mass_unit_(mass_unit), length_unit_(length_unit), temp_unit_(temp_unit),
+        rho_unit_(mass_unit_ / (length_unit_ * length_unit_ * length_unit_)),
+        freq_unit_(1. / time_unit_),
+        inv_emiss_unit_(length_unit_ * time_unit_ * time_unit_ / mass_unit_),
+        inv_num_emiss_unit_(length_unit_ * length_unit_ * length_unit_ *
+                            time_unit_) {}
```

Comment:
Possible someone would want to use a different "numbers" unit someday... like moles instead of actual number. Worth adding a `number_unit_`?

Reply:
Oh this is a good idea. Let's pass on this for now, but keep it in mind?